src/main/java/games/strategy/triplea/settings/ClientSetting.java

```diff
@@ -0,0 +1,144 @@
+package games.strategy.triplea.settings;
+
+import java.io.File;
+import java.util.prefs.BackingStoreException;
+import java.util.prefs.Preferences;
+
+import javax.swing.UIManager;
+
+import com.google.common.base.Strings;
+
+import games.strategy.debug.ClientLogger;
+import games.strategy.engine.ClientFileSystemHelper;
+
+/**
+ * List of settings that can be adjusted and stored with a Client's OS. On windows this would be the registry,
+ * these values survive game re-installs. These values can be made available for edit by adding a corresponding
+ * UI binding in {@code ClientSettingUiBinding}. Not all system settings will have UI bindings.
+ *
+ * <p>
+ * Note: After saving values, `ClientSetting.flush()` needs to be called to persist those values.
+ * </p>
+ *
+ *
+ * <p>
+ * Typical usage:
+ * </p><code><pre>
+ *
+ * // loading a value
+ * String value = ClientSetting.AI_PAUSE_DURATION.value();
+ *
+ * // saving value
+ * ClientSetting.AI_PAUSE_DURATION.save(500);
+ * ClientSetting.flush();
+ * </pre></code>
+ */
+public enum ClientSetting {
+  AI_PAUSE_DURATION(400),
+  ARROW_KEY_SCROLL_SPEED(70),
+  BATTLE_CALC_SIMULATION_COUNT_DICE(2000),
+  BATTLE_CALC_SIMULATION_COUNT_LOW_LUCK(5000),
+  CONFIRM_DEFENSIVE_ROLLS(false),
+  CONFIRM_ENEMY_CASUALTIES(false),
+  FASTER_ARROW_KEY_SCROLL_MULTIPLIER(2),
+  SPACE_BAR_CONFIRMS_CASUALTIES(true),
+  LOBBY_LAST_USED_HOST,
+  LOBBY_LAST_USED_PORT,
+  LOOK_AND_FEEL_PREF(UIManager.getSystemLookAndFeelClassName()),
+  MAP_EDGE_SCROLL_SPEED(30),
+  MAP_EDGE_SCROLL_ZONE_SIZE(30),
+  MAP_FOLDER_OVERRIDE,
+  MAP_LIST_OVERRIDE,
+  PROXY_CHOICE,
+  PROXY_HOST,
+  PROXY_PORT,
+  SAVE_GAMES_FOLDER_PATH(new File(ClientFileSystemHelper.getUserRootFolder(), "savedGames")),
+  SERVER_OBSERVER_JOIN_WAIT_TIME(180),
+  SERVER_START_GAME_SYNC_WAIT_TIME(180),
+  SHOW_BATTLES_WHEN_OBSERVING(true),
+  SHOW_BETA_FEATURES(false),
+  TEST_LOBBY_HOST,
+  TEST_LOBBY_PORT,
+  TRIPLEA_FIRST_TIME_THIS_VERSION_PROPERTY(true),
+  TRIPLEA_LAST_CHECK_FOR_ENGINE_UPDATE,
+  TRIPLEA_LAST_CHECK_FOR_MAP_UPDATES,
+  TRIPLEA_PROMPT_TO_DOWNLOAD_TUTORIAL_MAP,
+  TRIPLEA_SERVER_OBSERVER_JOIN_WAIT_TIME,
+  TRIPLEA_SERVER_START_GAME_SYNC_WAIT_TIME,
+  USER_MAPS_FOLDER_PATH(new File(ClientFileSystemHelper.getUserRootFolder(), "downloadedMaps")),
+  WHEEL_SCROLL_AMOUNT(60);
+
+  private final String defaultValue;
+
+
+  ClientSetting() {
+    this.defaultValue = "";
+  }
+
+  ClientSetting(final File file) {
+    this.defaultValue = file.getAbsolutePath();
```

Comment:
You should probably reuse other constructors here: `this(file.getAbsolutePath());`